docs/api/incoming-message.md

```diff
@@ -51,12 +51,17 @@ A `String` representing the HTTP status message.
 
 #### `response.headers`
 
-An `Record<string, string[]>` representing the response HTTP headers. The `headers` object is
+An `Record<string, string | string[]>` representing the response HTTP headers. The `headers` object is
 formatted as follows:
 
 * All header names are lowercased.
-* Each header name produces an array-valued property on the headers object.
-* Each header value is pushed into the array associated with its header name.
+* Duplicates of `age`, `authorization`, `content-length`, `content-type`,
+`etag`, `expires`, `from`, `host`, `if-modified-since`, `if-unmodified-since`,
+`last-modified`, `location`, `max-forwards`, `proxy-authorization`, `referer`,
+`retry-after`, `server`, or `user-agent` are discarded.
+* `set-cookie` is always an array. Duplicates are added to the array.
```

Comment:
So all entries are of type `string`, some with separators, but the only exception to the rule being `set-cookie` which is `string[]`?

Could be worth rewording to further clarify this.

Reply:
This description aligns with the documentation of Node's [`http.IncomingMessage`](https://nodejs.org/api/http.html#http_message_headers), I think it is probably enough.